aws/resource_aws_emr_instance_fleet.go

```diff
@@ -0,0 +1,193 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/emr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsEMRInstanceFleet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEMRInstanceFleetCreate,
+		Read:   resourceAwsEMRInstanceFleetRead,
+		Update: resourceAwsEMRInstanceFleetUpdate,
+		Delete: resourceAwsEMRInstanceFleetDelete,
+		Schema: map[string]*schema.Schema{
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"instance_fleet_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(emr.InstanceFleetType_Values(), false),
+			},
+			"instance_fleet": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem:     InstanceFleetConfigSchema(),
+			},
+		},
+	}
+}
+
+func resourceAwsEMRInstanceFleetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).emrconn
+
+	addInstanceFleetInput := &emr.AddInstanceFleetInput{
+		ClusterId: aws.String(d.Get("cluster_id").(string)),
+	}
+
+	if l := d.Get("instance_fleet").([]interface{}); len(l) > 0 && l[0] != nil {
+		addInstanceFleetInput.InstanceFleet = readInstanceFleetConfig(
+			l[0].(map[string]interface{}),
+			*aws.String(d.Get("instance_fleet_type").(string)))
+	}
+
+	log.Printf("[DEBUG] Creating EMR instance fleet params: %s", addInstanceFleetInput)
+	resp, err := conn.AddInstanceFleet(addInstanceFleetInput)
+	if err != nil {
+		return err
```

Comment:
We should return context with this error messaging for operators and code maintainers:

```suggestion
		return fmt.Errorf("error adding EMR Instance Fleet: %w", err)
```